browser/search_engines/search_engine_provider_service.cc

```diff
@@ -28,9 +28,24 @@ SearchEngineProviderService::SearchEngineProviderService(
       base::Bind(&SearchEngineProviderService::OnPreferenceChanged,
                  base::Unretained(this)));
 
-  auto data = TemplateURLPrepopulateData::GetPrepopulatedEngine(
-      otr_profile->GetPrefs(),
-      TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO);
+  std::vector<TemplateURLPrepopulateData::BravePrepopulatedEngineID>
+      alt_search_providers = {
+          TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO,
+          TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO_DE,
+          TemplateURLPrepopulateData::
+              PREPOPULATED_ENGINE_ID_DUCKDUCKGO_AU_NZ_IE
+      };
+  size_t index = 0;
+  std::unique_ptr<TemplateURLData> data = nullptr;
+
+  do {
+    data = TemplateURLPrepopulateData::GetPrepopulatedEngine(
+        otr_profile->GetPrefs(),
+        alt_search_providers[index++]);
+  } while (data == nullptr && index < alt_search_providers.size());
```

Comment:
How about brand-new c++11's range based for-loop? :)
```
  std::unique_ptr<TemplateURLData> data;
  for (const auto& id : alt_search_providers) {
    data = TemplateURLPrepopulateData::GetPrepopulatedEngine(
        otr_profile->GetPrefs(),
        id);
    if (data)
      break;
  }
```

Reply:
def a good call 😄 I the do/while was much cleaner than iterator based one - but the shorthand in C++11 in quite nice (sometimes, I'm stuck back in the C++98 days)